jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -173,6 +174,22 @@ public JibContainerBuilder createJibContainerBuilder(
 
       FileCollection snapshotDependencies =
           nonProjectDependencies.filter(file -> file.getName().contains("SNAPSHOT"));
+
+      FileCollection changingDependencies =
+          project.files(
+              project
+                  .getConfigurations()
+                  .getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME)
+                  .getDependencies()
+                  .stream()
+                  .filter(artifact -> artifact instanceof ExternalModuleDependency)
+                  .filter(
+                      moduleDependency ->
+                          ((ExternalModuleDependency) moduleDependency).isChanging())
+                  .collect(Collectors.toList()));
+
+      snapshotDependencies.plus(changingDependencies);
```

Comment:
Based on [this](https://github.com/GoogleContainerTools/jib/issues/694#issue-343698282), I feel like this will end up having duplicate entries if the filename contains `SNAPSHOT`. I'd say first we need to verify whether it is really true that a file containing `SNAPSHOT` sets the "changing" mark. And if so, probably we don't need to do `nonProjectDependencies.filter(file -> file.getName().contains("SNAPSHOT")` anymore and just put `isChanging` into the snapshot layer directly.

That said, this kind of uncertainty is one of many reasons we need to add tests. Could you look at `GradleProjectPropertiesTest` and think of some tests? We can't really merge code that is untested.

Reply:
I installed `jib-gradle-plugin:1.7.1-SNAPSHOT` to local maven and used in a sample project to verify that `SNAPSHOT` dependencies does not have `isChanging`.
I also added a test case to build the tar, extract layers and verify that both the `SNAPSHOT` jar and changing jar are in the same layer.

I also tried to build the project like below instead of mocking.
```
 Project layersProject = ProjectBuilder.builder().withProjectDir(new 
 File("test/resources/gradle/proejects/gradle-layers")).withName("root").build();
 GradleProjectProperties gradleProjectProperties = new GradleProjectProperties(project, 
 mockLogger);
    JavaContainerBuilder javaContainerBuilder =
            JavaContainerBuilder.from(RegistryImage.named("base"))
                    .setAppRoot(AbsoluteUnixPath.get("/app"))
                    .setModificationTimeProvider((ignored1, ignored2) -> 
                      SAMPLE_FILE_MODIFICATION_TIME);
   JibContainerBuilder jibContainerBuilder =  
gradleProjectProperties.createJibContainerBuilder(javaContainerBuilder,
DEFAULT_CONTAINERIZING_MODE);
 BuildConfiguration configuration =
            JibContainerBuilderTestHelper.toBuildConfiguration(
                    jibContainerBuilder,
                    Containerizer.to(RegistryImage.named("to"))
                            .setExecutorService(MoreExecutors.newDirectExecutorService()));
    ContainerBuilderLayers layers = new ContainerBuilderLayers(configuration);
```
But the project object was not completely built so didn't give the expected result.

Main problem I noticed while trying to write the unit tests is, the current `FileCollections` are all mocked but the FileCollection I am trying to build has to come from `DefaultExternalModuleDependency` object and I could not find a way to mock all the involved Gradle objects.

Bottom line, I could verify the functionality using external project and a tar build junit, but with those changes `com.google.cloud.tools.jib.gradle.GradleProjectPropertiesTest#testCreateContainerBuilder_correctFiles` test case is failing.
I need help.

Reply:
@rajovictor thanks for looking into the matter. About writing a unit test, I'll check out your test code in your comment and think of something, but it may take some time.

Reply:
BTW, I also had to make few changes to the way I was checking and building the changingDependencies collection but will commit once we get the junits working.

Reply:
I've looked into this, and the simplest way will be to add an integration test and forget about adding some in `GradleProjectProperties`.

`GradleLayerConfigurationIntegrationTest` builds a sophisticated sample project at `jib-gradle-plugin/src/test/resources/gradle/projects/all-local-multi-service` and verifies each file in the image layers. I think you could define some changing dependencies in `jib-gradle-plugin/src/test/resources/gradle/projects/all-local-multi-service/complex-service/build.gradle` and verify if they go into the SNAPSHOT layer.

To run integration tests, you will do `./gradlew integrationTest`. But for it to work, the `JIB_INTEGRATION_TESTING_LOCATION` environment variable should be defined first. See `CONTRIBUTING.md` for the details.

Reply:
Thank you for the reply.
After some attempts, I got the full tar and the container build (till layer construction) executed with successful dependency and snapshot verification.
But then due to the changes in `GradleProjectProperties` current Unit tests were failing with NPEs  on some of `filecollections`.
Lemme spend some more time on them to see if I can get them to working.